hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/Dispatch.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.workflows.handle.flow.dispatch;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.Key;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.node.app.service.token.records.FinalizeContext;
+import com.hedera.node.app.signature.KeyVerifier;
+import com.hedera.node.app.spi.fees.FeeAccumulator;
+import com.hedera.node.app.spi.fees.Fees;
+import com.hedera.node.app.spi.workflows.HandleContext;
+import com.hedera.node.app.workflows.TransactionInfo;
+import com.hedera.node.app.workflows.dispatcher.ReadableStoreFactory;
+import com.hedera.node.app.workflows.handle.record.RecordListBuilder;
+import com.hedera.node.app.workflows.handle.record.SingleTransactionRecordBuilderImpl;
+import com.hedera.node.app.workflows.handle.stack.SavepointStackImpl;
+import com.hedera.node.app.workflows.prehandle.PreHandleResult;
+import com.swirlds.platform.state.PlatformState;
+import com.swirlds.state.spi.info.NodeInfo;
+import java.time.Instant;
+import java.util.Set;
+
+/**
+ * The context needed for executing business logic of a service. This has two implementations - one for user transactions scope
+ * and one for dispatched child transactions scope.
+ */
+public interface Dispatch {
+    /**
+     * The builder for the transaction record in the scope
+     * @return the builder
+     */
+    SingleTransactionRecordBuilderImpl recordBuilder();
+
+    /**
+     * The fees calculated for the transaction
+     * @return the fees
+     */
+    Fees fees();
+
+    /**
+     * The transaction info for the transaction
+     * @return the transaction info
+     */
+    TransactionInfo txnInfo();
+
+    /**
+     * The payer of the transaction. This will be the synthetic payer for child transactions.
+     * @return the payer
+     */
+    AccountID syntheticPayer();
```

Comment:
I think the name is misleading as it is only a synthetic payer for child transactions. Maybe just `payer` or `transactionPayer`?